packages/playback-core/test/player.test.js

```diff
@@ -0,0 +1,169 @@
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+      },
+      video
+    );
+    assert.equal(typeof core, 'object');
+    assert.equal(typeof core.engine, 'object');
+  });
+
+  it('teardown', async function () {
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+      },
+      video
+    );
+    teardown(video, core);
+
+    assert(!video.hasAttribute('src'), 'src is removed on teardown');
+  });
+
+  it('handle error', async function () {
+    const video = document.createElement('video');
+    initialize(
+      {
+        src: 'https://mux.com/', // not working src url
+      },
+      video
+    );
+    // error event must be attached after initialize(), otherwise the
+    // native error will not be stopped propagating in playback core.
+    await new Promise((resolve) => video.addEventListener('error', resolve));
+
+    assert.equal(video.error.code, MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED);
+    assert.equal(video.error.message, '');
+
+    assert.equal(getError(video).code, MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED);
+    assert.equal(
+      getError(video).message,
+      'An unsupported error occurred. The server or network failed, or your browser does not support this format.'
+    );
+  });
+
+  it('setAutoplay("any")', async function () {
+    this.timeout(10000);
+
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+        autoplay: false,
+      },
+      video
+    );
+
+    core.setAutoplay('any');
+    await new Promise((resolve) => video.addEventListener('playing', resolve));
+
+    assert(!video.paused, 'is playing after core.setAutoplay("any")');
+  });
+
+  it('setAutoplay("muted")', async function () {
+    this.timeout(10000);
+
+    const video = document.createElement('video');
+    const core = initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+        autoplay: false,
+      },
+      video
+    );
+
+    core.setAutoplay('muted');
+    await new Promise((resolve) => video.addEventListener('playing', resolve));
+
+    assert(!video.paused, 'is playing after core.setAutoplay("muted")');
+    assert(video.muted, 'video is muted for autoplaying');
+  });
+
+  it('preload="none"', async function () {
+    this.timeout(10000);
+
+    const video = document.createElement('video');
+    initialize(
+      {
+        src: 'https://stream.mux.com/23s11nz72DsoN657h4314PjKKjsF2JG33eBQQt6B95I.m3u8',
+        preload: 'none',
+      },
+      video
+    );
+
+    assert.equal(video.preload, 'none', 'preload is none');
+    await aTimeout(3000);
```

Comment:
nitpick(non-blocking): I think we're testing too much here. It looks like you're trying to test:

1. that `preload="none"` will not load any data
2. that `preload="none"` will **_eventually_** load data by explicitly invoking `play()`.

The first test should probably have some event listeners that will cause the test to fail if they are invoked (e.g. `loadstart` and/or `loadedmetadata`) and passes if, after some amount of time (such as your example 3 seconds), its `readyState=0` (aka "HAVE_NOTHING").

The second test should have the same initialization and pass if the relevant event listeners (e.g. `loadeddata` and/or `canplay` are invoked after invoking `play()` (can use the `done()` function for this case

Reply:
good feedback will update in a follow up pr